multistep/commonsteps/step_http_server.go

```diff
@@ -54,14 +54,21 @@ func (s *StepHTTPServer) Run(ctx context.Context, state multistep.StateBag) mult
 		return multistep.ActionHalt
 	}
 
-	ui.Say(fmt.Sprintf("Starting HTTP server on port %d", s.l.Port))
+	httpAddr, present = state.GetOk("http_ip")
+	if present {
+		if httpAddr.(string) == "" {
+			httpAddr = s.HTTPAddress
+		}
+        }
```

Comment:
nit: on the nested if statements

```suggestion
	if present && httpAddr.(string) == "" {
			httpAddr = s.HTTPAddress
        }
```